include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1837 @@
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset & operator<<=(size_t const count) noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        uint8_t offset = 58 - size() + count;
+        data_.bits <<= offset;
+        data_.bits >>= offset - count;
+        return *this;
+    }
+
+    /*!\brief Performs binary shift right on the current object.
+     * \param[in] count Amount to shift to the right.
+     * \returns *this
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_right_inplace.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset & operator>>=(size_t const count) noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        data_.bits >>= count;
+        return *this;
+    }
+
+    /*!\brief Performs binary shift right.
+     * \param[in] count Amount to shift to the right.
+     * \returns Copy of *this with bits shifted count to the right.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_right_copy.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset operator>>(size_t const count) const noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        dynamic_bitset tmp{*this};
+        tmp >>= count;
+        return tmp;
+    }
+
+    /*!\brief Performs binary shift left.
+     * \param[in] count Amount to shift to the left.
+     * \returns Copy of *this with bits shifted count to the left.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_left_copy.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset operator<<(size_t const count) const noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        dynamic_bitset tmp{*this};
+        tmp <<= count;
+        return tmp;
+    }
+
+    /*!\brief Sets all bits to 1.
+     * \returns *this
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_set_all.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset & set() noexcept
+    {
+        data_.bits = ~0ULL >> 6;
```

Comment:
depending on the capacity?

Reply:
Good catch!
I think I'll refactor the test such that I test default capacity and a smaller capacity.